### PR TITLE
CUMULUS-2434: Update granules CMR metadata task to have both S3 and distribution links

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-2434: As a Cumulus integrator, I want the Cumulus to add CMR links to the S3 location](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2434)

## Changes

*  Updated `@cumulus/cmrjs` `updateCMRMetadata` functions to add both HTTPS URLS and S3 URIs to CMR metadata.
* Updated `update-granules-cmr-metadata-file-links` task to add both HTTPS URLs and S3 URIs to the OnlineAccessURLs field of CMR metadata.
* To use the updated `update-granules-cmr-metadata-file-links` task, the granule UMM-G metadata should have version 1.6.2 or later, since CMR s3 link type 'GET DATA VIA DIRECT ACCESS' is not valid until UMM-G version [1.6.2](https://cdn.earthdata.nasa.gov/umm/granule/v1.6.2/umm-g-json-schema.json)

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: packages/cmrjs/tests/cmr-utils/test-mergeURLs.js [hunk 1]
```diff
@@ -226,8 +226,8 @@ test('Does Complicated merging', (t) => {
     {
       URL: 's3://new/path/to/metadata.cmr.xml',
       URLDescription: 'from Updated',
-      Type: 'keeps Type: ORIGINAL',
-      MimeType: 'also from original: text/xml',
+      Type: 'GET DATA VIA DIRECT ACCESS',
```
Comment (flamingbear) [inline packages/cmrjs/tests/cmr-utils/test-mergeURLs.js:229]: Is this correct, why did the cmr.xml file get a "GET DATA VIA DIRECT ACCESS" type on it to start with, and should it have `Type: 'ORIGINAL Type'` still?

File: tasks/update-granules-cmr-metadata-file-links/tests/test-index.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ test.serial('Should update existing etag on CMR metadata file', async (t) => {
   t.not(newEtag, previousEtag);
 });
 
-test.serial('update-granules-cmr-metadata-file-links throws an error when cmr file type is distribution and no distribution endpoint is set', async (t) => {
+test.serial('update-granules-cmr-metadata-file-links throws an error when cmr file type is distribution/both and no distribution endpoint is set', async (t) => {
```
Comment (flamingbear) [inline tasks/update-granules-cmr-metadata-file-links/tests/test-index.js:125]: this should probably be just type is both, since that's what's being tested.
```suggestion
test.serial('update-granules-cmr-metadata-file-links throws an error when cmr file type is both and no distribution endpoint is set', async (t) => {
```

**Review by flamingbear — CHANGES_REQUESTED** [review]
I have one typo, one question and one suggestion.  Looks good overall.

**Comment by jennyhliu** [comment]
@flamingbear Can you review again when you get chance?  The integration tests should work after lzards issue has been fixed

File: packages/cmrjs/tests/cmr-utils/test-mergeURLs.js (hunk 1 above)
Comment (flamingbear) [reply packages/cmrjs/tests/cmr-utils/test-mergeURLs.js:229]: I'm not sure I fully understand, but I think I'll trust you because that function has gotten a bit complicated.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,18 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 
+### BREAKING CHANGES
+
+- **CUMULUS-2434**
+  - - To use the updated `update-granules-cmr-metadata-file-links` task, the granule UMM-G metadata should have version 1.6.2 or later, since CMR s3 link type 'GET DATA VIA DIRECT ACCESS' is not valid until UMM-G version [1.6.2](https://cdn.earthdata.nasa.gov/umm/granule/v1.6.2/umm-g-json-schema.json)
+
+### Changed
+- **CUMULUS-2434**
+  - Updated `@cumulus/cmrjs` `updateCMRMetadata` and related functions to add both HTTPS URLS and S3 URIs to CMR metadata.
+  - Updated `update-granules-cmr-metadata-file-links` task to add both HTTPS URLs and S3 URIs to the OnlineAccessURLs field of CMR metadata. The task configureation parameter `cmrGranuleUrlType` now has default value `both`.
```
Comment (flamingbear) [inline CHANGELOG.md:17]:
```suggestion
  - Updated `update-granules-cmr-metadata-file-links` task to add both HTTPS URLs and S3 URIs to the OnlineAccessURLs field of CMR metadata. The task configuration parameter `cmrGranuleUrlType` now has default value `both`.
```

File: tasks/move-granules/schemas/config.json [hunk 4]
```diff
@@ -37,12 +37,6 @@
       "type": "string",
       "description": "The api distribution endpoint"
     },
```
Comment (flamingbear) [inline tasks/move-granules/schemas/config.json:39]: Why did you remove this from the schema instead of renaming?

**Review by flamingbear — CHANGES_REQUESTED** [review]
One typo and one question about the schema.

File: tasks/move-granules/schemas/config.json (hunk 4 above)
Comment (jennyhliu) [reply tasks/move-granules/schemas/config.json:39]: cmrGranuleUrlType is not used in `move-granules`, the update metadata functionality has been moved to `update-granules-cmr-metadata-file-links`, do we still need to keep it?

Comment (flamingbear) [reply tasks/move-granules/schemas/config.json:39]: No you're right, I was completely spacing that this is a schema for the task.

**Review by flamingbear — APPROVED** [review]

**Comment by flamingbear** [comment]
Thanks for the updates, looks good to go. 👍
